lib/strategy.js

```diff
@@ -1,228 +1,58 @@
- *
- * Examples:
+ *                     'aq:name', 'email', 'phone', 'address', 'aq:location', 'aq:push' or none.
  *
- *     passport.use(new AuthentiqStrategy({
- *         clientID: 'Authentiq Client ID',
- *         clientSecret: 'Authentiq Client Secret'
- *         callbackURL: 'https://www.example.net/auth/authentiq/callback',
- *       },
- *       function(accessToken, refreshToken, profile, cb) {
- *         User.findOrCreate(..., function (err, user) {
- *           cb(err, user);
- *         });
- *       }
- *     ));
- *
- * @constructor
- * @param {object} options
- * @param {function} verify
- * @access public
+ * @param {Object} options
+ * @param {Function} verify
+ * @api public
  */
 function Strategy(options, verify) {
     this.options = xtend({}, options, {
-        authorizationURL: options.authorizationURL || 'https://connect.authentiq.io/authorize',
-        tokenURL: options.tokenURL || 'https://connect.authentiq.io/token',
-        userProfileURL: options.userProfileURL || 'https://connect.authentiq.io/userinfo',
-        scopeSeparator: ' ',
-        scope: options.scope.indexOf('openid') === -1 ? options.scope += " openid" : options.scope     // append openID if needed
+        // providerURL: options.providerURL || 'https://connect.authentiq.io',
+        sessionKey: options.sessionKey || 'authentiq:connect.authentiq.io',
     });
 
-    OAuth2Strategy.call(this, this.options, verify);
-
-    this.name = 'authentiq';
-
-    // this._oauth2.useAuthorizationHeaderforGET(true);
-
-    var self = this;
-
-    var _oauth2_getOAuthAccessToken = this._oauth2.getOAuthAccessToken;
-    this._oauth2.getOAuthAccessToken = function (code, params, callback) {
-        _oauth2_getOAuthAccessToken.call(self._oauth2, code, params, function (err, accessToken, refreshToken, params) {
-            if (err) {
-                return callback(err);
-            }
-            if (!accessToken) {
-                return callback({
-                    statusCode: 400,
-                    data: JSON.stringify(params)
-                });
-            }
-
-            // We're probably are already clear to continue with the ID token by parsing it but
-            //
-            // Call the callback with ID token and Access token so that we can call the /userinfo in case something goes wrong
-
-            var tokens = {};
-            tokens.accessToken = accessToken;
-
-            if (params.id_token) {
-                tokens.idToken = params.id_token;
-                callback(null, tokens, refreshToken, params);
-            } else {
-                callback(null, accessToken, refreshToken, params);
-            }
+    // pass a custom Resolver, that ignores WebFinger
+    this.options.resolver = this.options.resolver || new Resolver(this.options);
+
+    // return back the Authentiq application's Client, based on the options
+    function getClientCallback(issuer, cb) {
+        return cb(null, {
```

Comment:
@skion the base package supports auto-client-registration, removing this function allows us to use it. For now, we disable this functionality and use the Client that the developer has set.